### PR TITLE
CDK: Raise error on passing unsupported value formats as query parameters

### LINKED ISSUE #25032: [Low-code CDK] Raise error on passing unsupported value formats as query parameters
## Context

As discussed in [this slack thread](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1681165885984729?thread_ts=1681159224.845049&cid=C02TL38U5L7), the url library used in the low-code CDK doesn't support passing arrays as query parameters, and there is no standard way to format objects in query parameters.

Therefore, the low-code CDK should throw an explicit and helpful error message if an array or object is passed into a query parameter, so that the user isn't confused as to why this doesn't work.

(Note: the low-code CDK should throw this error instead of the builder UI so that behavior is consistent across both, and because the builder cannot tell if an interpolated value would result in an array or object.)

## Alternatives considered
Alternatively we could switch to a different library that supports arrays as query params, and decide on a format to use for objects in query params or let the developer decide that. 

However, since this is an edge case that we haven't seen strong demand for, and deciding on a format is not very helpful if the API doesn't support it, then we are opting for an error instead.

## Acceptance Criteria
- An error is thrown by the low-code CDK if an array or object is attempted to be passed in as a query parameter to the request

## Open question
- Should the exception be thrown from HttpStream or should this be scoped to low-code?

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Resolve https://github.com/airbytehq/airbyte/issues/25032 bug.*

## How
*Raise error inside `_request_params` for `HttpRequester` if value is array or object.*

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py`
2. `airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py`

### Airbyter

Before merging:
- Pull Request description explains what problem it is solving
- Code change is unit tested
- Build and my-py check pass
- Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- PR is reviewed and approved
      
After merging:
- [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py [hunk 1]
```diff
@@ -342,52 +342,84 @@ def test_send_request_params(provider_params, param_params, authenticator_params
...
+    assert sent_request.url.split("?", 1)[-1] == expected_query_params
+
+
+@pytest.mark.parametrize(
+    "request_parameters, config, invalid_value_for_key",
+    [
         pytest.param(
             {"k": "[1,2]"},
             {},
-            "k=1&k=2",
```
Comment (girarda) [inline airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py:348]: why are these test cases removed?

**Review by girarda — COMMENTED** [review]
confirmed this works as expected by modifying the greenhouse connector to do the following:
- added `an_array` field to the spec defined as an array
- added a request parameter to the http_requester:
```
    request_parameters:
      input_val: "{{ config['an_array'] }}"
```

File: airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py:348]: If I understood correctly, according to the acceptance criterion, `An error is thrown by the low-code CDK if an array or object is attempted to be passed in as a query parameter to the request`, now will not be able to pass this type of data.
Currently, in the master branch, user can pass an array of strings/int/float, and in this case we will have the following examples:
1. `{"k": "[1,2]"} = "k=1&k=2"`
2.  `{"k": ["param1", "param2"]} = "k=param1&k=param2"`
3. `{"k": [12.55, 13.48]} = "k=12.55&k=13.48"`

If we want to keep the ability to pass an array of strings/int/float, but only throw an error if an array of arrays or an array of objects is passed, you need to modify the check conditions a bit.
In this case, we will get an error only if we pass it: 
1. `{"k": [[1,2], [3, 4]]} = ValueError`
2. `{"k": [{3: 4}]} = ValueError`

And objects will throw an error in any case.
`{"k": {"param1": "some_value"}} = ValueError`

**Review by girarda — APPROVED** [review]
:shipit:

**Pull request merged**